tester/src/test/scala/spinal/tester/code/FormalApi.scala

```diff
@@ -0,0 +1,418 @@
+package spinal.tester.code
+
+import spinal.core._
+import spinal.core.formal._
+import spinal.tester._
+
+/** Test the functional equivalence of the two formal backends:
+  * - SymbiYosys: generates SystemVerilog with SVA statements
+  * - GHDL:       generates VHDL with PSL statements
+  * 
+  * The functionality is slowly tested by building ontop of previous tested
+  * functionality tests. If multiple tests fail, the "higher" one is likely the
+  * culprit.
+  * 
+  * Tested API:
+  *  - [ ] past
+  *  - [ ] pastValid
+  *  - [ ] pastValidAfterReset
+  *  - [ ] rose
+  *  - [ ] fell
+  *  - [ ] changed
+  *  - [ ] stable
+  *  - [ ] initstate
+  *  - [x] anyseq / anyconst / allseq / allconst
+  *  - [x] assumeInitial
+  *  - [x] assume
+  *  - [x] cover
+  *  - [x] assert
+  *  - [ ] QOL show backtrace
+  *  - [ ] assume/assert inside conditional scope
+  *  - [x] assert during reset
+  *  - [x] assert outside of reset
+  *  - [ ] multiple clock domains
+  */
+class FormalApiTest extends SpinalFormalFunSuite {
```

Comment:
How about keep the xxxxxxFunSuite naming pattern for test suite?

Reply:
What do you mean? This is not a class meant to be used as `FunSuite`. It is just a testing class where multiple `test()` calls are made. The current naming I have borrowed from [`BitVectorTests.scala#L31`](https://github.com/SpinalHDL/SpinalHDL/blob/499d6dca89c7c3b38cd9de1d5f4412d8166a7252/tester/src/test/scala/spinal/core/BitVectorTests.scala#L31) where it is `xxxxxxTest`.

Reply:
Ok, I missunderstand it a API that wrap for verifications which could inherit from.